prow/plugins/testfreeze/testfreeze.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testfreeze
+
+import (
+	"fmt"
+	"html/template"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+	"k8s.io/test-infra/prow/plugins/testfreeze/checker"
+)
+
+const (
+	PluginName                  = "testfreeze"
+	defaultKubernetesBranch     = "master"
+	defaultKubernetesRepoAndOrg = "kubernetes"
+	templateString              = `{{ if .InTestFreeze -}}
+Please note that we're already in [Test Freeze](https://github.com/kubernetes/sig-release/blob/master/releases/release_phases.md#test-freeze) for the ` + "`{{ .Branch }}`" + ` branch. This means every merged PR has to be cherry-picked into the release branch to be part of the upcoming {{ .Tag }} release.
+{{ end }}`
+)
+
+func init() {
+	plugins.RegisterPullRequestHandler(PluginName, handlePullRequestEvent, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: fmt.Sprintf(
+			"The %s plugin adds additional documentation about cherry-picks during the Test Freeze period.",
+			PluginName,
+		),
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Featured:  false,
+		WhoCanUse: "Anyone",
+	})
+	return pluginHelp, nil
+}
+
+func handlePullRequestEvent(p plugins.Agent, e github.PullRequestEvent) error {
+	h := newHandler()
+	log := p.Logger
+	if err := h.handle(
+		log,
+		p.GitHubClient,
+		e.Action,
+		e.Number,
+		e.Repo.Owner.Login,
+		e.Repo.Name,
+		e.PullRequest.Base.Ref,
+	); err != nil {
+		log.WithError(err).Error("skipping")
+	}
+	return nil
+}
+
+type handler struct {
+	impl impl
+}
+
+func newHandler() *handler {
+	return &handler{
+		impl: &defaultImpl{},
+	}
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate . impl
+type impl interface {
+	CheckInTestFreeze(*logrus.Entry) (*checker.Result, error)
+	CreateComment(plugins.PluginGitHubClient, string, string, int, string) error
+}
+
+type defaultImpl struct{}
```

Comment:
nit: `impl` and `defaultImpl` are not informative names. Why not

```go
type Checker struct {
  checker
}

type checker interface{/*...*/}

type defaultChecker struct {/*...*/}
```
?

Reply:
Yes, changed that for the `checker` package. For the `testfreeze` package I had to choose a different name (`verifier`) to not clash with the `checker` package.